### PR TITLE
Ip address selection policy

### PR DESCRIPTION
**Pull request by jlporter**
Add ipAddressSeelectionPolicy field to BackendService and RegionBackendService.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: added `ipAddressSelectionPolicy` field to `google_compute_backend_service` and `google_compute_region_backend_service`.
```

### CODE REVIEW

**Review by roaks3 — COMMENTED** [review]
It looks like there may still be activity on this (and some failing tests), so just let me know when this is ready for a review.

**Comment by jlporter** [comment]
Thanks @roaks3, this should be ready for review. I believe the failing tests are unrelated to my change.

**Comment by jlporter** [comment]
Made one more update, as this field should be available in v1 now, but should be ready for review now.

File: mmv1/templates/terraform/examples/backend_service_ip_address_selection_policy.tf.tmpl [hunk 1]
```diff
@@ -0,0 +1,5 @@
+resource "google_compute_backend_service" "{{$.PrimaryResourceId}}" {
```
Comment (roaks3) [inline mmv1/templates/terraform/examples/backend_service_ip_address_selection_policy.tf.tmpl:1]: I don't seen these examples being picked up anywhere (to be used in the docs and/or tests). Did you perhaps forget to add `examples` blocks in the yaml files?

**Review by roaks3 — COMMENTED** [review]
Code LGTM, but might be missing tests here

File: mmv1/templates/terraform/examples/backend_service_ip_address_selection_policy.tf.tmpl (hunk 1 above)
Comment (jlporter) [reply mmv1/templates/terraform/examples/backend_service_ip_address_selection_policy.tf.tmpl:1]: Ah, indeed, they should be added now.

File: mmv1/templates/terraform/examples/region_backend_service_ip_address_selection_policy.tf.tmpl [hunk 2]
```diff
@@ -0,0 +1,18 @@
+resource "google_compute_region_backend_service" "{{$.PrimaryResourceId}}" {
+  name                            = "{{index $.Vars "region_backend_service_name"}}"
+  region                          = "us-central1"
+  health_checks                   = [google_compute_region_health_check.default.id]
```
Comment (roaks3) [inline mmv1/templates/terraform/examples/region_backend_service_ip_address_selection_policy.tf.tmpl:4]: Looks like this should be `google_compute_region_health_check. health_check.id`, then hopefully tests will pass

**Review by roaks3 — CHANGES_REQUESTED** [review]

File: mmv1/templates/terraform/examples/region_backend_service_ip_address_selection_policy.tf.tmpl (hunk 2 above)
Comment (jlporter) [reply mmv1/templates/terraform/examples/region_backend_service_ip_address_selection_policy.tf.tmpl]: Thanks, should be fixed. I figured out how to run manually and it's passing in my repo now also.
```
$ make testacc TEST=./google/services/compute TESTARGS='-run=TestAccComputeRegionBackendService_regionBackendServiceIpAddressSelectionPolicyExample'
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google/services/compute -v -run=TestAccComputeRegionBackendService_regionBackendServiceIpAddressSelectionPolicyExample -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google/version.ProviderVersion=acc"
=== RUN   TestAccComputeRegionBackendService_regionBackendServiceIpAddressSelectionPolicyExample
=== PAUSE TestAccComputeRegionBackendService_regionBackendServiceIpAddressSelectionPolicyExample
=== CONT  TestAccComputeRegionBackendService_regionBackendServiceIpAddressSelectionPolicyExample
--- PASS: TestAccComputeRegionBackendService_regionBackendServiceIpAddressSelectionPolicyExample (40.34s)
PASS
ok      github.com/hashicorp/terraform-provider-google/google/services/compute  40.804s
```

**Review by roaks3 — APPROVED** [review]

**Pull request merged**